internal/resource/http.go

```diff
@@ -247,3 +253,21 @@ func (c HttpClient) getReaderWithHeader(url string, header http.Header) (io.Read
 		}
 	}
 }
+
+// addSystemInfoHeaders will add an ID, a VERSION_ID (if present), and a
+// VARIANT_ID (if present) header to the client's ignition config GET request
+func addSystemInfoHeaders(header *http.Header) error {
+	or, err := osrelease.New()
+	if err != nil {
+		return err
+	}
+
+	header.Add("ID", or.ID)
+	if len(or.VERSION_ID) != 0 { // optional field
```

Comment:
It's preferable to write the following instead:

```go
if or.VERSION_ID != "" {
```

This makes it much more clear that the type is a string and you are looking for an empty one.

Reply:
fixed